src/main/java/quotify_app/data_access/DBUserDataAccessObject.java

```diff
@@ -0,0 +1,109 @@
+package quotify_app.data_access;
+
+import quotify_app.entities.User;
+import quotify_app.entities.UserFactory;
+import quotify_app.usecases.login.LoginUserDataAccessInterface;
+import quotify_app.usecases.signup.SignupUserDataAccessInterface;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+
+/**
+ * Data access object for user-related operations, interacting with an SQLite database.
+ */
+public class DBUserDataAccessObject implements SignupUserDataAccessInterface, LoginUserDataAccessInterface {
+
+    private final UserFactory userFactory;
+    private String currentUsername;
+
+    public DBUserDataAccessObject(UserFactory userFactory) {
+        this.userFactory = userFactory;
+        initializeDatabase();
+    }
+
+    /**
+     * Initializes the database and ensures that the users table exists.
+     */
+    private void initializeDatabase() {
+        try (Connection conn = DatabaseConnection.getConnection();
+             PreparedStatement stmt = conn.prepareStatement(
+                     "CREATE TABLE IF NOT EXISTS users (" +
+                             "username TEXT PRIMARY KEY, " +
+                             "email TEXT UNIQUE, " +
+                             "password TEXT NOT NULL)"
+             )) {
+            stmt.executeUpdate();
+            System.out.println("Database initialized successfully. Users table created if it didn't exist.");
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
```

Comment:
Need to look a tiny bit more into using postman for this, my experience with it is mainly like making manual calls to endpoints. If we were to make tests that made use of postman's API, I'd have to figure that out

Reply:
We can talk about it today when we meet up because database testing is essential to know if we are interacting with the DB in the way we anticipate we are.